front/lib/rate_limiter.ts

```diff
@@ -0,0 +1,30 @@
+import { v4 as uuidv4 } from "uuid";
+
+import { redisClient } from "@app/lib/redis";
+
+export async function rateLimiter(
+  key: string,
+  maxPerTimeframe: number,
+  timeframeSeconds: number
+) {
+  const redis = await redisClient();
+  const redisKey = `rate_limiter:${key}`;
+  try {
+    const zcountRes = await redis.zCount(
+      redisKey,
+      new Date().getTime() - timeframeSeconds * 1000,
+      "+inf"
+    );
+    const remaining = maxPerTimeframe - zcountRes;
+    if (remaining > 0) {
+      await redis.zAdd(redisKey, {
```

Comment:
Is this equivalent to the INCR that is suggested here: https://redis.com/glossary/rate-limiting/ ?

What's the difference between zCount / zAdd and GET/INCR?

Reply:
With GET/INCR you would get a count over the minute number X.
With zAdd/zCount you get a count of the last X seconds.

Reply:
Ok

Reply:
>With GET/INCR you would get a count over the minute number X.

Complexity is O(1)

>With zAdd/zCount you get a count of the last X seconds.

Complexity is O(log(n)); n = number of calls that went through for a given rate limit key.